### ISSUE TITLE
Delete when destroyed

### ISSUE THREAD

**Issue by bibbybabby** [studio-minus/ppg-bugs-and-requests#2322]
**Feature description**
Context option, when selected the object is deleted when it is destroyed by any means

**Feature purpose**
Eco-friendly contraptions that don't leave a fucking mess that always fucking gets in the fucking way and causes lag

**Additional context**
hehe bomb go boom > prop gone

**Comment by DindinYT37** [comment]
How do you expect a contraption to know if it's destroyed, exactly?

**Comment by bibbybabby** [comment]
> How do you expect a contraption to know if it's destroyed, exactly?

When a part is destroyed

**Comment by DindinYT37** [comment]
> > How do you expect a contraption to know if it's destroyed, exactly?
> 
> When a part is destroyed

Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".

**Comment by bibbybabby** [comment]
> > > How do you expect a contraption to know if it's destroyed, exactly?
> > 
> > 
> > When a part is destroyed
> 
> Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".

When it takes sufficient damage

**Comment by DindinYT37** [comment]
> > > > How do you expect a contraption to know if it's destroyed, exactly?
> > > 
> > > 
> > > When a part is destroyed
> > 
> > 
> > Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".
> 
> When it takes sufficient damage

So, you just expect it to magically know?

**Comment by bibbybabby** [comment]
> > > > > How do you expect a contraption to know if it's destroyed, exactly?
> > > > 
> > > > 
> > > > When a part is destroyed
> > > 
> > > 
> > > Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".
> > 
> > 
> > When it takes sufficient damage
> 
> So, you just expect it to magically know?

How do you think every destructible object works?

**Comment by DindinYT37** [comment]
> > > > > > How do you expect a contraption to know if it's destroyed, exactly?
> > > > > 
> > > > > 
> > > > > When a part is destroyed
> > > > 
> > > > 
> > > > Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".
> > > 
> > > 
> > > When it takes sufficient damage
> > 
> > 
> > So, you just expect it to magically know?
> 
> How do you think every destructible object works?

How do you think every non-destructible modded object will work?

**Comment by bibbybabby** [comment]
> > > > > > > How do you expect a contraption to know if it's destroyed, exactly?
> > > > > > 
> > > > > > 
> > > > > > When a part is destroyed
> > > > > 
> > > > > 
> > > > > Yeah, but how do you expect something like for example a metal cube to know when it was "destroyed".
> > > > 
> > > > 
> > > > When it takes sufficient damage
> > > 
> > > 
> > > So, you just expect it to magically know?
> > 
> > 
> > How do you think every destructible object works?
> 
> How do you think every non-destructible modded object will work?

Based on size,mass and material

**Comment by DindinYT37** [comment]
> > > How do you think every destructible object works?
> > 
> > 
> > How do you think every non-destructible modded object will work?
> 
> Based on size,mass and material

Yeah, that will totally work out perfectly as you imagine it...
